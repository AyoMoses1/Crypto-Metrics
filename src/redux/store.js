import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './cryptos/cryptoSlice';

const store = configureStore({
  reducer: {
    cryptoReducer,
  },
});

export default store;
