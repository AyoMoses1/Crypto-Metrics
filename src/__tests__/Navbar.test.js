import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import NavBar from '../components/Navbar';
import store from '../redux/store';

it('NavBar component test', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
