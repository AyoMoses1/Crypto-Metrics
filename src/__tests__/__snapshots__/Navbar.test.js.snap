// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component test 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Navbar scroll
    </a>
    <button
      aria-controls="navbarScroll"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="navbarScroll"
    >
      <div
        className="me-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
        onKeyDown={[Function]}
        style={
          Object {
            "maxHeight": "100px",
          }
        }
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          Books
        </a>
      </div>
      <form
        className="d-flex"
      >
        <div
          className="icons-con"
        >
          <i
            aria-hidden="true"
            className="fa fa-microphone"
          />
          <i
            aria-hidden="true"
            className="fa fa-cog"
          />
        </div>
        <input
          aria-label="Search"
          className="me-2 form-control"
          onChange={[Function]}
          placeholder="Search"
          type="search"
        />
        <button
          className="btn btn-outline-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</nav>
`;
